modules/javafx.graphics/src/main/java/com/sun/glass/ui/WindowControlsOverlay.java

```diff
@@ -0,0 +1,539 @@
+import javafx.beans.value.ObservableValue;
+import javafx.css.CssMetaData;
+import javafx.css.PseudoClass;
+import javafx.css.SimpleStyleableBooleanProperty;
+import javafx.css.SimpleStyleableIntegerProperty;
+import javafx.css.SimpleStyleableObjectProperty;
+import javafx.css.StyleConverter;
+import javafx.css.Styleable;
+import javafx.css.StyleableBooleanProperty;
+import javafx.css.StyleableIntegerProperty;
+import javafx.css.StyleableObjectProperty;
+import javafx.css.StyleableProperty;
+import javafx.geometry.Dimension2D;
+import javafx.geometry.HPos;
+import javafx.geometry.HorizontalDirection;
+import javafx.geometry.NodeOrientation;
+import javafx.geometry.VPos;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.scene.layout.Region;
+import javafx.scene.paint.Paint;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
+import javafx.util.Subscription;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * Contains the visuals and behaviors for the minimize/maximize/close buttons on an {@link StageStyle#EXTENDED}
+ * window for platforms that use client-side decorations (Windows and Linux/GTK). This control supports
+ * left-to-right and right-to-left orientations, as well as a customizable layout order of buttons.
+ *
+ * <h2>Substructure</h2>
+ * <ul>
+ *     <li>{@link Region} — {@code window-button-container}
+ *     <ul>
+ *         <li>{@link Region} — {@code window-button}, {@code minimize-button}
+ *         <li>{@link Region} — {@code window-button}, {@code maximize-button}
+ *         <li>{@link Region} — {@code window-button}, {@code close-button}
+ *     </ul>
+ * </ul>
+ *
+ * <table style="white-space: nowrap">
+ *     <caption>CSS properties of {@code window-button-container}</caption>
+ *     <thead>
+ *         <tr><th>CSS property</th><th>Values</th><th>Default</th><th>Comment</th></tr>
+ *     </thead>
+ *     <tbody>
+ *         <tr>
+ *             <th>-fx-button-placement</th><td>[ left | right ]</td><td>right</td>
+ *             <td style="white-space: break-line">
+ *                 Specifies the placement of the window buttons on the left or the right side of the window.
+ *             </td>
+ *         </tr>
+ *         <tr>
+ *             <th>-fx-allow-rtl</th><td>&lt;boolean&gt;</td><td>true</td>
+ *             <td style="white-space: break-line">
+ *                 Specifies whether the minimize/maximize/close buttons support right-to-left orientations.
+ *             </td>
+ *         </tr>
+ *     </tbody>
+ * </table>
+ *
+ * <table style="white-space: nowrap">
+ *     <caption>CSS properties of {@code minimize-button}, {@code maximize-button}, {@code close-button}</caption>
+ *     <thead>
+ *         <tr><th>CSS property</th><th>Values</th><th>Default</th><th>Comment</th></tr>
+ *     </thead>
+ *     <tbody><tr>
+ *         <th>-fx-button-order</th><td>&lt;integer&gt;</td><td>0/1/2</td>
```

Comment:
The buttons are sorted with `List.sort`, so that will dictate what happens in this scenario. Note that this class is not API, and neither users nor the platform will have any control over it.

Reply:
Any value will work. This is the javadoc for `buttonOrder`:

        /**
         * Specifies the layout order of this button relative to the other buttons.
         * Buttons with a lower value are laid out before buttons with a higher value.
         * <p>
         * This property corresponds to the {@code -fx-button-order} CSS property.
         */